checkpoint.go

```diff
@@ -78,12 +79,30 @@ checkpointed.`,
 	},
 }
 
-func getCheckpointImagePath(context *cli.Context) string {
+func prepareImagePaths(context *cli.Context) (string, string, error) {
 	imagePath := context.String("image-path")
+
 	if imagePath == "" {
 		imagePath = getDefaultImagePath(context)
 	}
-	return imagePath
+
+	if err := os.MkdirAll(imagePath, 0755); err != nil {
+		return "", "", err
+	}
+
+	parentPath := context.String("parent-path")
+
+	if parentPath == "" {
+		return imagePath, parentPath, nil
+	}
+
+	realParent := filepath.Join(imagePath, parentPath)
```

Comment:
I guess you should check first if `parentPath` is absolute, and error out if it's not
```go
if !filepath.IsAbs(parentPath) {
   return errors.New("--parent-path must be relative")
}
```
or so.